### PR TITLE
[Project] link to contract/analytic account

### PR DESCRIPTION
**Pull request by atp-odoo**
[IMP] project,project_issue,project_issue_sheet,rating: add project_type field in project and add related field of type in task and issue,
[REM]:  removed is_visible_happy_customer field from rating.

Task: https://www.odoo.com/web?#id=19609&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.TUBtj27YXVQaF1GU

### CODE REVIEW

File: addons/project/project.py [hunk 1]
```diff
@@ -113,6 +113,18 @@ def _get_alias_models(self, cr, uid, context=None):
         """ Overriden in project_issue to offer more options """
         return [('project.task', "Tasks")]
 
+    def _count_analytic_accounts(self, cr, uid, ids, field_name, arg, context):
+        return {
+            project_id: len(self.browse(cr, uid, project_id, context=context).analytic_account_id)
+            for project_id in ids
```
Comment (atp-odoo) [inline addons/project/project.py]: is it really required ? analytic_account_id is many2one field for project. so we always get only one analytic account for single project.

File: addons/project/project.py [hunk 2]
```diff
@@ -113,6 +113,18 @@ def _get_alias_models(self, cr, uid, context=None):
...
+        }
+
+    def action_open_analytic_accounts(self, cr, uid, ids, context=None):
+        res = self.pool.get('ir.actions.act_window').for_xml_id(
```
Comment (atp-odoo) [inline addons/project/project.py]: use self.pool['ir.actions.act_window']

File: addons/project/project.py [hunk 3]
```diff
@@ -396,11 +358,14 @@ def onchange_planned(self, cr, uid, ids, planned=0.0, effective=0.0):
...
     def onchange_project(self, cr, uid, id, project_id, context=None):
+        vals = {'partner_id': False}
         if project_id:
             project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
-            if project and project.partner_id:
-                return {'value': {'partner_id': project.partner_id.id}}
-        return {'value': {'partner_id': False}}
+            vals.update({
+                'project_type': project and project.project_type,
+                'partner_id': project and project.partner_id.id
```
Comment (atp-odoo) [inline addons/project/project.py]: no need to check project.partner_id condition ? if we don't have project.partner_id then ?

File: addons/project/project.py [hunk 4]
```diff
@@ -478,6 +443,7 @@ def _compute_displayed_image(self, cr, uid, ids, prop, arg, context=None):
         'user_email': fields.related('user_id', 'email', type='char', string='User Email', readonly=True),
         'attachment_ids': fields.one2many('ir.attachment', 'res_id', domain=lambda self: [('res_model', '=', self._name)], auto_join=True, string='Attachments'),
         'displayed_image_id': fields.function(_compute_displayed_image, relation='ir.attachment', type="many2one", string='Attachment'),
+        'project_type': fields.related('project_id', 'project_type', relation='project.project', type='char', string='Project Type'),
```
Comment (atp-odoo) [inline addons/project/project.py]: no need to give relation field. project_type is simple char field.

File: addons/project/project_view.xml [hunk 5]
```diff
@@ -166,10 +167,7 @@
             <field name="arch" type="xml">
                 <search string="Search Project">
                     <field name="name" string="Project Name"/>
-                    <filter string="Open" name="Current" domain="[('state', '=','open')]"/>
-                    <filter string="Pending" name="Pending" domain="[('state', '=','pending')]"/>
-                    <filter string="Template" name="Template" domain="[('state', '=','template')]"/>
-                    <separator/>
+                    <filter string="Member" name="Member" domain="['|',('user_id', '=', uid),('members', '=', uid)]"/>
```
Comment (atp-odoo) [inline addons/project/project_view.xml]: check with rebase member is not exist in view.

File: addons/project_issue/project_issue.py [hunk 6]
```diff
@@ -120,11 +120,14 @@ def _compute_day(self, cr, uid, ids, fields, args, context=None):
...
     def on_change_project(self, cr, uid, ids, project_id, context=None):
+        vals = {'partner_id': False}
         if project_id:
             project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
-            if project and project.partner_id:
-                return {'value': {'partner_id': project.partner_id.id}}
-        return {'value': {'partner_id': False}}
+            vals.update({
+                'project_type': project and project.project_type,
+                'partner_id': project and project.partner_id.id
```
Comment (atp-odoo) [inline addons/project_issue/project_issue.py]: same as above. check project.partner_id.

File: addons/project_issue/project_issue.py [hunk 7]
```diff
@@ -184,6 +187,7 @@ def on_change_project(self, cr, uid, ids, project_id, context=None):
         'user_email': fields.related('user_id', 'email', type='char', string='User Email', readonly=True),
         'date_action_last': fields.datetime('Last Action', readonly=1),
         'date_action_next': fields.datetime('Next Action', readonly=1),
+        'project_type': fields.related('project_id', 'project_type', relation='project.project', type='char', string='Project Type'),
```
Comment (atp-odoo) [inline addons/project_issue/project_issue.py]: no need to define relation. it's char field.

**Comment by antonylesuisse** [comment]
copy_with task should be added as an action with an ir_values (no javascript).

**Comment by atp-odoo** [comment]
Hello @antonylesuisse ,
we already tried for this solution with ir_values. but 
if we do with ir_values then two problem occurs : 
1) in Form view we can duplicate record with task but in save mode. we cannot able to duplicate record with edit mode at that time. then if i want to edit that record we have to switch on list view and then need to search for that records.
2) that "duplicate with task" appeared In more option of list view also. so multiple record can be duplicated. its become wrong.

**Comment by hmo-odoo** [comment]
cancalled

**Pull request closed without merging**